vega_sim/scenario/common/agents.py

```diff
@@ -3528,3 +3528,54 @@ def finalise(self):
 
     def name(self):
         return self._agent.name()
+
+
+class SettlementDataSubmitter(StateAgentWithWallet):
+    """
+    Submits pre-supplied oracle data every at every step
+    """
+
+    NAME_BASE = "oracle_data_submitter"
+
+    def __init__(
+        self,
+        key_name: str,
+        oracle_data_generator: Iterable[float],
+        market_name: Optional[str] = None,
+        tag: str = "",
+        wallet_name: Optional[str] = None,
+    ):
+        super().__init__(wallet_name=wallet_name, key_name=key_name, tag=tag)
+        self.current_step = 0
+        self.market_name = market_name
+        self.oracle_data_generator = oracle_data_generator
+
+        self.prev_data = None
+        self.curr_data = None
+
+    def initialise(
+        self,
+        vega: Union[VegaServiceNull, VegaServiceNetwork],
+        create_key: bool = True,
+        mint_key: bool = True,
+    ):
+        super().initialise(vega=vega, create_key=create_key, mint_key=mint_key)
+        self.market_id = self.vega.find_market_id(name=self.market_name)
+
+    def step(self, vega_state: VegaState):
+        self.current_step += 1
+        self.prev_data = self.curr_data
+        self.curr_data = next(self.oracle_data_generator)
+
+        # depending on order in which agents are traversed the market may not be initialised yet
+        if self.market_id is None:
+            id = self.vega.find_market_id(name=self.market_name)
+            if id is None:
+                return
+            self.market_id = id
+
+        self.vega.submit_settlement_data(
+            settlement_key=self.key_name,
+            settlement_price=self.curr_data,
+            market_id=self.market_id,
+        )
```

Comment:
Maybe add an argument for how frequently this actually submits so you can play around with that?

Reply:
Good point, added now